src/app/pages/signin-page/signin-page.component.ts

```diff
@@ -1,8 +1,63 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, FormControl, Validators, AbstractControl } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import * as userActions from '../../core/store/user/user.actions';
+import { EmailFormatValidator } from '../../shared/validators/email-format';
+import { PasswordTrimValidator } from '../../shared/validators/password-match';
+import { selectReasonError } from '../../core/store/user/user.selectors';
 
 @Component({
   selector: 'app-signin-page',
   templateUrl: './signin-page.component.html',
   styleUrl: './signin-page.component.scss',
 })
-export class SigninPageComponent {}
+export class SigninPageComponent implements OnInit {
+  public loginForm: FormGroup = new FormGroup({});
+
+  public firstContact: boolean = false;
+
+  public error$: Observable<string>;
+
+  constructor(
+    private readonly FB: FormBuilder,
+    private store: Store,
+    private readonly router: Router,
+  ) {
+    this.error$ = this.store.select(selectReasonError);
+  }
+
+  ngOnInit(): void {
+    this.loginForm = this.FB.group({
```

Comment:
Please, make this formBuilder nonNullable like this.formBuilder.nonNullable.group()

Reply:
done